### PR TITLE
Support larger jwt tokens

### LINKED ISSUE #215: Investigate JWT size limitations in staging
After adding a user to about 50 projects 3 problems occured:

1) nginx started throwing: "upstream sent too big header while reading response header from upstream", because the cookie from .NET was too big
2) nginx start returning "400 Request Header Or Cookie Too Large", because the auth header from the client was too big
2) .NET started splitting the cookie into "chunks" (multiple suffixed cookies) to make sure it fits into browser limits and SvelteKit didn't know what to do with them

50 projects is quite a few. The project codes are also very large in this example. But 50 seems like a reasonable and safe limit to support for the time being. Ultimately our JWT is still a fair bit bulkier than it needs to be. e.g. "ProjectId" vs "Id" or "Manager" vs. simply "M".

<details><summary>Here's what is in the JWT token</summary>
<p>

{
  "sub": "cf430ec9-e721-450a-b6a1-9a853212590b",
  "email": "admin@test.com",
  "name": "Test Admin",
  "role": "admin",
  "proj": [
    {
      "Code": "test-race-flex--45cf1c7d",
      "Role": "Manager",
      "ProjectId": "45cf1c7d-e82e-42c2-b0ab-bed897d49299"
    },
    {
      "Code": "test-race-flex--2772cdce",
      "Role": "Manager",
      "ProjectId": "2772cdce-442b-4a20-8891-772075f8ce8e"
    },
    {
      "Code": "test-race-flex--0fbaec21",
      "Role": "Manager",
      "ProjectId": "0fbaec21-b6de-414f-9491-2a6ca2697608"
    },
    {
      "Code": "test-race-flex--cba4dc7f",
      "Role": "Manager",
      "ProjectId": "cba4dc7f-e438-4a9c-9293-e6603627f479"
    },
    {
      "Code": "test-race-flex--9dd00a35",
      "Role": "Manager",
      "ProjectId": "9dd00a35-23eb-4940-a1f8-0e11fd688922"
    },
    {
      "Code": "test-race-flex--02af5f6d",
      "Role": "Manager",
      "ProjectId": "02af5f6d-6937-4aa6-8e4c-86c2c2596a78"
    },
    {
      "Code": "test-race-flex--374cf36c",
      "Role": "Manager",
      "ProjectId": "374cf36c-ac1c-45b8-987b-9438e74db58f"
    },
    {
      "Code": "test-race-flex--1813f15f",
      "Role": "Manager",
      "ProjectId": "1813f15f-34ab-40c3-b03f-b57eb74f2efd"
    },
    {
      "Code": "test-race-flex--0afee859",
      "Role": "Manager",
      "ProjectId": "0afee859-b599-499b-bf3c-06c761eb9756"
    },
    {
      "Code": "test-race-flex--552ea565",
      "Role": "Manager",
      "ProjectId": "552ea565-0f6b-4b5e-b084-6b28c9cae7e6"
    },
    {
      "Code": "test-race-flex--bbe80b3e",
      "Role": "Manager",
      "ProjectId": "bbe80b3e-120f-42a3-b823-aeef58b72cd6"
    },
    {
      "Code": "test-race-flex--47a2de2a",
      "Role": "Manager",
      "ProjectId": "47a2de2a-8bca-446d-8266-7e1591d81902"
    },
    {
      "Code": "test-race-flex--fe914ad7",
      "Role": "Manager",
      "ProjectId": "fe914ad7-048e-4ff5-9fd3-a3fbe7d18849"
    },
    {
      "Code": "test-race-flex--53ca7e96",
      "Role": "Manager",
      "ProjectId": "53ca7e96-1039-45dd-8bb8-f634d51197e5"
    },
    {
      "Code": "test-race-flex--75343078",
      "Role": "Manager",
      "ProjectId": "75343078-c416-4702-9d73-99534f02081b"
    },
    {
      "Code": "test-race-flex--555af26a",
      "Role": "Manager",
      "ProjectId": "555af26a-e145-426a-96e2-885ce4d23e58"
    },
    {
      "Code": "test-race-flex--0e61e62c",
      "Role": "Manager",
      "ProjectId": "0e61e62c-3ac2-4e00-ab09-d726b31951d1"
    },
    {
      "Code": "test-race-flex--f117a6bc",
      "Role": "Manager",
      "ProjectId": "f117a6bc-c24a-453a-83bd-e5967d719288"
    },
    {
      "Code": "test-race-flex--f92238d8",
      "Role": "Manager",
      "ProjectId": "f92238d8-fe75-4035-8ba6-c9299e697749"
    },
    {
      "Code": "test-race-flex--e5fda91b",
      "Role": "Manager",
      "ProjectId": "e5fda91b-968d-471a-bc51-bd4ebd385184"
    },
    {
      "Code": "test-race-flex--96a04338",
      "Role": "Manager",
      "ProjectId": "96a04338-009f-41f8-9ae0-1f3a7da11d2a"
    },
    {
      "Code": "test-race-flex--e7050933",
      "Role": "Manager",
      "ProjectId": "e7050933-4a25-4b43-b790-ae3faef63b41"
    },
    {
      "Code": "test-race-flex--648d6bab",
      "Role": "Manager",
      "ProjectId": "648d6bab-c5ba-4069-be2e-15f543257f12"
    },
    {
      "Code": "test-race-flex--0e528c0c",
      "Role": "Manager",
      "ProjectId": "0e528c0c-21ef-4abc-9eb2-02683bbef658"
    },
    {
      "Code": "test-race-flex--fc9a629b",
      "Role": "Manager",
      "ProjectId": "fc9a629b-9b94-4bd4-a7ef-108c88d02c18"
    },
    {
      "Code": "test-race-flex--b94154f7",
      "Role": "Manager",
      "ProjectId": "b94154f7-bf77-49d3-941a-804e622e04b8"
    },
    {
      "Code": "test-race-flex--6bf1e5c9",
      "Role": "Manager",
      "ProjectId": "6bf1e5c9-65c5-4208-8279-d529b6fba91d"
    },
    {
      "Code": "test-race-flex--2c4b0a99",
      "Role": "Manager",
      "ProjectId": "2c4b0a99-d250-4667-88b9-74d56351abd2"
    },
    {
      "Code": "test-race-flex--ab71699f",
      "Role": "Manager",
      "ProjectId": "ab71699f-7a97-46af-b040-e067f790b6e7"
    },
    {
      "Code": "test-race-flex--a4aab8c8",
      "Role": "Manager",
      "ProjectId": "a4aab8c8-d10d-4b9f-b60d-0e816b2078ce"
    },
    {
      "Code": "test-race-flex--f9ad8a49",
      "Role": "Manager",
      "ProjectId": "f9ad8a49-cc42-4158-8ebe-b92cb66d24c0"
    },
    {
      "Code": "test-race-flex--b1058ed7",
      "Role": "Manager",
      "ProjectId": "b1058ed7-7a7b-443e-a485-3cfe0209f153"
    },
    {
      "Code": "test-race-flex--809489f3",
      "Role": "Manager",
      "ProjectId": "809489f3-0804-43c2-a151-bac3714d225b"
    },
    {
      "Code": "test-race-flex--e541e111",
      "Role": "Manager",
      "ProjectId": "e541e111-3383-433e-a605-13eb3f51b766"
    },
    {
      "Code": "test-race-flex--af91865a",
      "Role": "Manager",
      "ProjectId": "af91865a-ed8f-4209-aab2-a25010fc9e55"
    },
    {
      "Code": "test-race-flex--1bb658d2",
      "Role": "Manager",
      "ProjectId": "1bb658d2-40db-4f94-b074-c1c6cdd836c7"
    },
    {
      "Code": "test-race-flex--2838169c",
      "Role": "Manager",
      "ProjectId": "2838169c-5895-4144-b254-7fa11e92783a"
    },
    {
      "Code": "test-race-flex--b1327b6c",
      "Role": "Manager",
      "ProjectId": "b1327b6c-95f4-4a2d-9247-ef1f924a54c8"
    },
    {
      "Code": "test-race-flex--0f97db88",
      "Role": "Manager",
      "ProjectId": "0f97db88-c48c-4593-98de-5d27888fc54f"
    },
    {
      "Code": "test-race-flex--1a4d5e48",
      "Role": "Manager",
      "ProjectId": "1a4d5e48-af63-46e9-b382-4bad42c39556"
    },
    {
      "Code": "test-race-flex--dfc0dfea",
      "Role": "Manager",
      "ProjectId": "dfc0dfea-8a8f-45e1-9d7d-691ad3ca0eeb"
    },
    {
      "Code": "test-race-flex--4f286f59",
      "Role": "Manager",
      "ProjectId": "4f286f59-6e42-4a36-9c04-005c03594096"
    },
    {
      "Code": "test-race-flex--46121b68",
      "Role": "Manager",
      "ProjectId": "46121b68-ae46-4d1f-b8a6-bfca6ece2137"
    },
    {
      "Code": "test-race-flex--783b988d",
      "Role": "Manager",
      "ProjectId": "783b988d-f3bd-44f5-b96d-5638bc619caa"
    },
    {
      "Code": "test-race-flex--0f4bee28",
      "Role": "Manager",
      "ProjectId": "0f4bee28-0491-43e8-b024-aa4676d1e4e0"
    },
    {
      "Code": "test-race-flex--26a2d8e7",
      "Role": "Manager",
      "ProjectId": "26a2d8e7-f74b-4bec-b49f-33df19cd9ce1"
    },
    {
      "Code": "test-race-flex--725d84c0",
      "Role": "Manager",
      "ProjectId": "725d84c0-4a03-4b4b-a1f6-dce5674917cc"
    },
    {
      "Code": "test-race-flex--f9957450",
      "Role": "Manager",
      "ProjectId": "f9957450-4f51-4573-a7e5-0432b96cda68"
    },
    {
      "Code": "test-race-flex--941c4c62",
      "Role": "Manager",
      "ProjectId": "941c4c62-c08a-440d-8f4d-cb1907d3bcd9"
    },
    {
      "Code": "test-race-flex--774a1a09",
      "Role": "Manager",
      "ProjectId": "774a1a09-f91c-40bd-b626-0cbbd8328fe5"
    },
    {
      "Code": "test-race-flex--71bc9eb3",
      "Role": "Manager",
      "ProjectId": "71bc9eb3-c039-4327-b7db-e7611b05e629"
    },
    {
      "Code": "test-race-flex--1c2dfee6",
      "Role": "Manager",
      "ProjectId": "1c2dfee6-35cc-4ef2-8fb2-7cd311dc0691"
    },
    {
      "Code": "test-race-flex--c427a31b",
      "Role": "Manager",
      "ProjectId": "c427a31b-cfef-475a-8b4a-c6820256efde"
    },
    {
      "Code": "test-race-flex--0de82f53",
      "Role": "Manager",
      "ProjectId": "0de82f53-840f-4cf4-8417-cfed906bcecf"
    },
    {
      "Code": "test-race-flex--93f16510",
      "Role": "Manager",
      "ProjectId": "93f16510-729d-4d79-84bc-e8ab99d9ed77"
    }
  ],
  "props.persistent": "",
  "props.issued": "Tue, 01 Aug 2023 09:07:12 GMT",
  "props.expires": "Wed, 16 Aug 2023 09:07:12 GMT",
  "nbf": 1690880832,
  "exp": 1692176832,
  "iat": 1690880832,
  "iss": "lexbox-api",
  "aud": "lexbox-api"
}

</p>
</details> 

[ingress-nginx docs: Accommodation for JWT](https://github.com/kubernetes/ingress-nginx/blob/d712dd9d92f1f46eab5f55d4643f3a45a7a97f90/docs/examples/customization/jwt/README.md)

This is improved in #216, but some of those changes only targeted dev, so there's presumably still work to do for staging/prod if we want larger token limits.

### PR DESCRIPTION
**Pull request by myieye**
Solves part of #215, but there's likely still work to do for staging/prod.

### CODE REVIEW

**Review by hahn-kev — COMMENTED** [review]
looks good. Merge when ready.

File: deployment/dev/ingress-deployment.yaml [hunk 1]
```diff
@@ -321,6 +321,9 @@ subjects:
 apiVersion: v1
 data:
   allow-snippet-annotations: "true"
+  proxy-buffer-size: "16k"
+  large-client-header-buffers: "4 16k"
```
[the remark this answers is not in the document]
Comment (myieye) [reply deployment/dev/ingress-deployment.yaml]: So, I managed to get this into the base kustomization.
If nothing else it was good exercise.
My minimal understanding of ingress and k8s is telling me that it's not going to have any effect in staging 🙁.

[the remark this answers is not in the document]
Comment (myieye) [reply deployment/dev/ingress-deployment.yaml]: @hahn-kev Did you say you were going to talk to Tim about this? That would be good. It seems silly to merge this this way.

[the remark this answers is not in the document]
Comment (hahn-kev) [reply deployment/dev/ingress-deployment.yaml]: I will when I get the chance.

**Comment by hahn-kev** [comment]
@myieye did you get a chance to test out staging and see how it responds with a big jwt? I'm not sure that staging will just automatically pickup the ConfigMap that you defined, so I'm not sure if it's any better.

**Comment by myieye** [comment]
@hahn-kev No, I haven't tested anything in staging yet. 

For the record, the ingress-nginx docs address our exact scenario: they have a [readme page](https://github.com/kubernetes/ingress-nginx/blob/d712dd9d92f1f46eab5f55d4643f3a45a7a97f90/docs/examples/customization/jwt/README.md) about increasing the proxy-buffer for large JWT tokens.

**Comment by myieye** [comment]
@hahn-kev So I:
- Removed my config map from /base
- Used the available annotation and 
- Reworded and edited the issue to no longer be a bug for MVP, but an engineering issue for v1

**Pull request merged**